src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useDispatch, useSelector } from 'react-redux';
+import Modal from "../../common/modal/Modal";
+import ModalTheme from '../ledger/ModalTheme';
+import MobileActionSheet from '../../common/modal/MobileActionSheet';
+import FormButton from '../../common/FormButton';
+import { Translate } from 'react-localize-redux';
+import TwoFactorVerifyInput from './TwoFactorVerifyInput';
+import { verifyTwoFactor, clearAlert, resendTwoFactor } from '../../../actions/account';
+import { wallet } from '../../../utils/wallet';
+
+const Form = styled.form`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    width: 100%;
+`
+
+const TwoFactorVerifyModal = ({ open, onClose }) => {
+
+    const [method, setMethod] = useState();
+    const [code, setCode] = useState('');
+    const dispatch = useDispatch();
+    const account = useSelector(({ account }) => account);
+    const loading = account.actionsPending.includes('VERIFY_TWO_FACTOR');
+    const error = account.requestStatus && account.requestStatus.messageCode === 'account.verifyTwoFactor.error';
+    // console.log(account)
+
+    useEffect(() => {
+        let isMounted = true;
+
+        const handleGetTwoFactor = async () => {
+            setMethod(await wallet.get2faMethod())
+        };
+
+        if (isMounted) {
+            handleGetTwoFactor()
+        }
+        
+        return () => { isMounted = false }
+    }, []);
+
+    const handleVerifyCode = async () => {
+        const { error } = await dispatch(verifyTwoFactor(null, code))
```

Comment:
seems like 2FA actions should be taking accountId and/or 2FA request ID as parameter to make it easier to avoid consistency problems when switching accounts